main.go

```diff
@@ -240,6 +260,68 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Start a goroutine to handle termination.
+	go func() {
+		// Cancel the main context when we are done.
+		defer cancel()
+
+		// Wait for a signal.
+		<-sigHandler.Done()
+
+		// Check if we need to do any cleanup.
+		client := mgr.GetClient()
+		instance := &v1.Installation{}
+		retries := 0
+		for {
+			if err := client.Get(ctx, utils.DefaultInstanceKey, instance); errors.IsNotFound(err) {
+				// No installation - we can exit immediately.
+				return
+			} else if err != nil {
+				// Error querying - retry after a small sleep.
+				if retries >= 5 {
+					log.Errorf("Too many retries, exiting with error: %s", err)
+					return
+				}
+				log.Errorf("Error querying Installation, will retry: %s", err)
+				retries++
+				time.Sleep(1 * time.Second)
+				continue
+			}
+
+			// Success
+			break
+		}
+
+		if instance.DeletionTimestamp == nil {
+			// Installation isn't terminating, so we can exit immediately.
+			return
+		}
+
+		// We need to wait for termination to complete. We can do this by checking if the Installation
+		// resource has been cleaned up or not.
+		to := 60 * time.Second
```

Comment:
Should we set `terminationGracePeriodSeconds` to 60s to match this since the default is 30s?